src/main/java/cristiano/Ronaldo.java

```diff
@@ -0,0 +1,194 @@
+        default:
+            break;
+        }
+    }
+
+    /**
+     * This method handles the marking and unmarking of a task/goal.
+     * The valid format to successfully mark or unmark a goal is: [command] [integer]
+     * Where command = mark/unmark, followed by a single space, followed by a valid integer.
+     * If an invalid format is used, a NumberFormatException message will be prompted.
+     * If the integer provided is out of bounds, an IndexOutOfBoundsException message will be prompted.
+     * If the user tries to mark or unmark a goal that has already been marked or unmarked,
+     * this method from the chatbot will remind the user its goal completion status.
+     * Goals completed will be marked as 'X'.
+     *
+     * @param input The goal index to be marked or unmarked;
+     */
+    public void handleGoal(String[] input) {
+        try {
+            if (input.length <= 1) {
+                reject("Mark/Unmark");
+                return;
+            }
+            int taskNumber = Integer.parseInt(input[1]) - 1;
+            Goal goal = goals.get(taskNumber);
+            if (input[0].equals("mark")) {
+                goal.markAsDone(this);
+            } else if (input[0].equals("unmark")) {
+                goal.markAsUndone(this);
+            }
+        } catch (NumberFormatException e) {
+            reject("Mark/Unmark");
+        } catch (IndexOutOfBoundsException e) {
+            reject("Range");
+        }
+    }
+
+    /**
+     * Prints out all the goals added, including its type and completion status.
+     * The goal index starts from '1', instead of the common array index, '0'.
+     */
+    public void showListOfGoals() {
+        System.out.println("Here are the goals to complete in order for you to reach your dreams:");
+        for (int i = 0; i < goals.size(); i++) {
+            Goal goal = goals.get(i);
+            System.out.println(i + 1 + "." + goal);
+        }
+        System.out.println(); //Indentation
+    }
+
+    public void printGoalCount() {
+        if (goals.size() == 1) {
+            System.out.println("Now you have " + goals.size() + " goal in the list.\n");
+        } else {
+            System.out.println("Now you have " + goals.size() + " goals in the list.\n");
+        }
+    }
+
+    public void addEvent(String input) {
+        String[] parts = input.split("/from | /to ", 3);
+        try {
+            String description = parts[0];
+            String from = parts[1];
+            String to  = parts[2];
+            if (description.trim().isEmpty() || from.trim().isEmpty() || to.trim().isEmpty()) {
+                reject("Event");
+                return;
+            }
+            Event event = new Event(description, from, to);
+            goals.add(event);
+            System.out.println("GOALLL! Your event has been added: \n" + event + "\n");
+            printGoalCount();
+        } catch (IndexOutOfBoundsException e) {
+            reject("Event");
+        }
```

Comment:
You may consider extracting this part of code and pass in parameters to differentiate events/dealines etc. as these few methods are pretty similar.